include/stl2/detail/iterator/basic_iterator.hpp

```diff
@@ -726,6 +726,15 @@ STL2_OPEN_NAMESPACE {
 			++*this;
 		}
 
+		constexpr basic_iterator& operator++(int) &
+		noexcept(noexcept(++declval<basic_iterator&>()))
+		requires !cursor::Input<C>() &&
+			 !cursor::PostIncrement<C>()
+		{
+			++*this;
+                        return *this;
+		}
```

Comment:
The fix; post-increment must return a self reference rather than a copy for output iterators that hold state.